### PR TITLE
HSEARCH-4116 Ignore empty, blank or null configuration properties when detecting unused properties

### PR DESCRIPTION
**Pull request by yrodiere**
https://hibernate.atlassian.net/browse/HSEARCH-4116

### CODE REVIEW

File: util/internal/integrationtest/backend/elasticsearch/src/main/java/org/hibernate/search/util/impl/integrationtest/backend/elasticsearch/ElasticsearchTestHostConnectionConfiguration.java [hunk 1]
```diff
@@ -53,20 +54,20 @@ private ElasticsearchTestHostConnectionConfiguration() {
...
 		properties.put( "username", username );
 		properties.put( "password", password );
-		properties.put( "aws.signing.enabled", String.valueOf( awsSigningEnabled ) );
+		properties.put( "aws.signing.enabled", awsSigningEnabled == null ? null : awsSigningEnabled.toString() );
 		properties.put( "aws.region", awsRegion );
 		properties.put( "aws.credentials.type", awsCredentialsType );
 		properties.put( "aws.credentials.access_key_id", awsCredentialsAccessKeyId );
 		properties.put( "aws.credentials.secret_access_key", awsCredentialsSecretAccessKey );
-		if ( awsSigningEnabled ) {
+		if ( awsSigningEnabled != null && awsSigningEnabled ) {
```
Comment (fax4ever) [inline util/internal/integrationtest/backend/elasticsearch/src/main/java/org/hibernate/search/util/impl/integrationtest/backend/elasticsearch/ElasticsearchTestHostConnectionConfiguration.java:70]: Maybe you can reuse `isAws()`. It looks the same.

**Review by fax4ever — APPROVED** [review]
Looks good. Just a comment

File: util/internal/integrationtest/backend/elasticsearch/src/main/java/org/hibernate/search/util/impl/integrationtest/backend/elasticsearch/ElasticsearchTestHostConnectionConfiguration.java (hunk 1 above)
Comment (yrodiere) [reply util/internal/integrationtest/backend/elasticsearch/src/main/java/org/hibernate/search/util/impl/integrationtest/backend/elasticsearch/ElasticsearchTestHostConnectionConfiguration.java:70]: Right; it should actually be just `awsSigningEnabled != null`.

Comment (yrodiere) [reply util/internal/integrationtest/backend/elasticsearch/src/main/java/org/hibernate/search/util/impl/integrationtest/backend/elasticsearch/ElasticsearchTestHostConnectionConfiguration.java:70]: Actually I was looking in the wrong place; you're absolutely right.

**Comment by yrodiere** [comment]
Merged, thanks!

**Pull request merged**